public/utils/storage-utils.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+
+const PREFIX = 'opendistro::security';
+
+export function getValue<T>(storage: Storage, category: string, key: string): T | null {
+  if (!storage) {
+    return null;
+  }
+  const item = storage.getItem(`${PREFIX}::${category}::${key}`);
+  if (!item) {
+    return null;
+  }
+  return JSON.parse(item) as T;
+}
+
+export function setValue(storage: Storage, category: string, key: string, value: any) {
+  if (!storage) {
+    return;
+  }
+  storage.setItem(`${PREFIX}::${category}::${key}`, JSON.stringify(value));
+}
+
+/*
+ * For cross page toasts
+ */
+
+const TOAST_CATEGORY = 'toast';
+
+export function setCrossPageToast(targetUrl: string, toast: Toast) {
+  setValue(sessionStorage, TOAST_CATEGORY, targetUrl, toast);
```

Comment:
Where is sessionStorage defined? Same for other places where this variable is used.

Reply:
It is actually window.sessionStorage and window can be omitted.